### PR TITLE
GQL Status objects for notifications

### PR DESCRIPTION
**Pull request by thelonelyvulpes**
This change introduces a preview of GQL-compliant notifications and status objects for the driver.

https://www.gqlstandards.org/

This change also introduces bolt 5.6& 5.5

### CODE REVIEW

File: Neo4j.Driver/Neo4j.Driver.Tests.TestBackend/Protocol/Result/SummaryJsonSerializer.cs [hunk 1]
```diff
@@ -162,22 +166,22 @@ private static object MapToPlanJson(IPlan plan)
         };
     }
 
-    private static object CreateNotificationList(IResultSummary summary)
+    private static object MapNotifications(IList<INotification> notifications, BoltProtocolVersion serverProtocol)
```
Comment (StephenCathcart) [inline Neo4j.Driver/Neo4j.Driver.Tests.TestBackend/Protocol/Result/SummaryJsonSerializer.cs]: Is serverProtocol needed?

Comment (thelonelyvulpes) [reply Neo4j.Driver/Neo4j.Driver.Tests.TestBackend/Protocol/Result/SummaryJsonSerializer.cs]: no, I'll undo that change

File: Neo4j.Driver/Neo4j.Driver/Internal/Protocol/BoltProtocolMessageFactory.cs [hunk 2]
```diff
@@ -85,11 +85,6 @@ internal class BoltProtocolMessageFactory : IBoltProtocolMessageFactory
         return new RunWithMetadataMessage(connection.Version, query, notificationsConfig: notificationsConfig);
     }
 
-    public TelemetryMessage NewTelemetryMessage(QueryApiType apiUsage)
```
Comment (StephenCathcart) [inline Neo4j.Driver/Neo4j.Driver/Internal/Protocol/BoltProtocolMessageFactory.cs:88]: Has this been moved elsewhere?

Comment (thelonelyvulpes) [reply Neo4j.Driver/Neo4j.Driver/Internal/Protocol/BoltProtocolMessageFactory.cs:88]: it is an unused method, there is a different NewTelemetry method

Comment (thelonelyvulpes) [reply Neo4j.Driver/Neo4j.Driver/Internal/Protocol/BoltProtocolMessageFactory.cs:88]: line 167 is the correct method

**Review by StephenCathcart — APPROVED** [review]
Feature looks good to me 👍

File: Neo4j.Driver/Neo4j.Driver/Internal/MessageHandling/Metadata/GqlStatusObjectsAndNotifications.cs [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+                cursorMetadata.ResultHadRecords switch
+                {
+                    true => GqlStatusObject.Success,
+                    false when cursorMetadata.ResultHadKeys => GqlStatusObject.NoData,
+                    _ => GqlStatusObject.OmittedResult
+                })
+            .OrderBy(
+                x => GetStatusOrderingValue(x) switch
+                {
+                    "02" => 0,
```
Comment (AndyHeap-NeoTech) [inline Neo4j.Driver/Neo4j.Driver/Internal/MessageHandling/Metadata/GqlStatusObjectsAndNotifications.cs:49]: Maybe a comment to explain the reason for the mappings in this switch.  Why 02 is 0 etc etc.

File: Neo4j.Driver/Neo4j.Driver/Internal/MessageHandling/Metadata/GqlStatusObjectsAndNotificationsCollector.cs [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+
+    private static IGqlStatusObject ConvertNotificationValuesToStatus(IDictionary<string, object> notification)
+    {
+        var severity = notification.GetValue<string>("severity", null);
+        var isWarning = string.Equals(
+            severity,
+            nameof(NotificationSeverity.Warning),
+            StringComparison.InvariantCultureIgnoreCase);
+        var status = isWarning
+            ? "01N42"
```
Comment (AndyHeap-NeoTech) [inline Neo4j.Driver/Neo4j.Driver/Internal/MessageHandling/Metadata/GqlStatusObjectsAndNotificationsCollector.cs:116]: Maybe reference or link to doc where value is specified. GQL spec or something perhaps?

File: Neo4j.Driver/Neo4j.Driver/Internal/Result/Summary/DatabaseInfo.cs [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+        : this(null)
+    {
+    }
+
+    public DatabaseInfo(string name)
+    {
+        Name = name;
+    }
+
+    public string Name { get; }
```
Comment (AndyHeap-NeoTech) [inline Neo4j.Driver/Neo4j.Driver/Internal/Result/Summary/DatabaseInfo.cs:30]: Just a personal style comment really. I prefer properties and member variables to be at the top of the class before the methods for ease of parsing.

**Review by AndyHeap-NeoTech — APPROVED** [review]
All looks good. Just a couple of very minor comments that don't really need anything doing if you can't be bothered.

File: Neo4j.Driver/Neo4j.Driver/Internal/Result/Summary/DatabaseInfo.cs (hunk 5 above)
Comment (thelonelyvulpes) [reply Neo4j.Driver/Neo4j.Driver/Internal/Result/Summary/DatabaseInfo.cs:30]: fields > constructors > properties > methods is our coding standard

**Pull request merged**